web/src/features/Auth/components/SecureAdminConsole.tsx

```diff
@@ -1,12 +1,39 @@
 import * as React from "react";
 import Helmet from "react-helmet";
-import { Utilities, dynamicallyResizeText } from "../utilities/utilities";
-import Loader from "./shared/Loader";
-import ErrorModal from "./modals/ErrorModal";
+import { Utilities, dynamicallyResizeText } from "@src/utilities/utilities";
+import Loader from "@src/components/shared/Loader";
+import ErrorModal from "@src/components/modals/ErrorModal";
 import "../scss/components/Login.scss";
-
-class SecureAdminConsole extends React.Component {
-  constructor(props) {
+import { App } from "@features/App";
+import { RouteComponentProps } from "react-router-dom";
+
+type Props = {
+  appName: string | null;
+  fetchingMetadata: boolean;
+  onLoginSuccess: () => Promise<App[]>;
+  pendingApp: () => Promise<App>;
+  checkIsHelmManaged: () => Promise<void>;
+  logo: string | null;
+} & RouteComponentProps;
+
+type State = {
+  password: string;
+  loginErr: boolean;
+  loginErrMessage: string;
+  authLoading: boolean;
+  loginInfo: {
+    method: string;
+  } | null;
+};
+type LoginResponse = {
+  token: string;
+  expires?: number;
+  sessionRoles: string;
+};
+class SecureAdminConsole extends React.Component<Props, State> {
+  loginText: React.RefObject<HTMLDivElement>;
```

Comment:
can you explain this? I'm learning :)

Reply:
`loginText` gets assigned in the constructor to a ref: 

````
this.loginText = React.createRef();
```

For some reason TS wouldn't implicitly type it so I had to explicitly type it with this statement. 

this assigns a type to `this.loginText` but without any assignment. I think TS removes this statement when things get compiled so it doesn't do anything in the js.

Reply:
thanks so much! i actually just ran into this in a file I'm working on!! super helpful!